Tables.jl integration (#17)
bhftbootcamp/TimeArrays.jl: docs/src/pages/tables.md

- **Schema Information**: Automatic type inference and column naming
- **Round-trip Conversion**: Convert to other table formats and back without data loss

### Column Structure

When viewed as a table, a `TimeArray` has two columns:
- `:timestamp` - Contains the time index values
- `:value` - Contains the corresponding data values

## Usage Examples

### Basic Table Interface

```julia
using TimeArrays, Dates
import Tables

# Create a TimeArray
timestamps = [DateTime("2024-01-01"), DateTime("2024-01-02"), DateTime("2024-01-03")]
values = [1.0, 2.0, 3.0]
ta = TimeArray(timestamps, values)

# Check Tables.jl compatibility
Tables.istable(ta)  # true

# Get schema information
schema = Tables.schema(ta)
schema.names  # (:timestamp, :value)
schema.types  # (DateTime, Float64)
```

### Column Access

```julia
# Access as columns
cols = Tables.columns(ta)
timestamps = Tables.getcolumn(cols, :timestamp)
values = Tables.getcolumn(cols, :value)
```

### Row Access

```julia
# Iterate over rows
for row in Tables.rows(ta)
    timestamp = Tables.getcolumn(row, :timestamp)
    value = Tables.getcolumn(row, :value)
    println("$timestamp: $value")
end
```

### Creating TimeArrays from Tables

> Use the `timearray_from_table` function to create TimeArrays from any Tables.jl-compatible source:

Reviewer: ```suggestion
Use the `TimeArray` constructor to create TimeArrays from any Tables.jl-compatible source:
```

```julia
# From a table with default column names
table_data = Tables.columntable([
    (timestamp=DateTime("2024-01-01"), value=10.0),
    (timestamp=DateTime("2024-01-02"), value=20.0)
])

> ta = timearray_from_table(table_data)

Reviewer: ```suggestion
ta = TimeArray(table_data)
```

# From a table with custom column names
custom_data = Tables.columntable([
    (time=DateTime("2024-01-01"), price=100.0),
    (time=DateTime("2024-01-02"), price=200.0)
])

> ta = timearray_from_table(custom_data; timestamp=:time, value=:price)

Reviewer: ```suggestion
ta = TimeArray(custom_data; timestamp=:time, value=:price)
```

```

### Round-trip Conversions

```julia
# TimeArray → Table → TimeArray
original = TimeArray(timestamps, values)
table_format = Tables.columntable(original)

> restored = timearray_from_table(table_format)

Reviewer: ```suggestion
restored = TimeArray(table_format)
```

# Data is preserved
original == restored  # true
```

## Integration with DataFrames.jl

```julia
using DataFrames

# TimeArray to DataFrame
df = DataFrame(ta)

# DataFrame to TimeArray

> ta_from_df = timearray_from_table(df)

Reviewer: ```suggestion
ta_from_df = TimeArray(df)
```

```

## Integration with CSV.jl

```julia
using CSV

# Save TimeArray to CSV
CSV.write("data.csv", ta)

# Load TimeArray from CSV

> ta_from_csv = timearray_from_table(CSV.File("data.csv"))

Reviewer: ```suggestion
ta_from_csv = TimeArray(CSV.File("data.csv"))
```
